### PR TITLE
Fix/2568/remove gz from filenames

### LINKED ISSUE #2568: Remove `.gz` from filenames
# Bug

## Expected Behavior

GIVEN a compressed file is loaded into the visualization

WHEN looking at the filename

THEN the `.gz` part is not visible (similar to `.cc.json`. That is also hidden)

## Actual Behavior

`.gz` is visible in multiple places.

## Screenshots

image

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json&file=codecharta_analysis.cc.json): yes
- CodeCharta Version: 1.85.0
- OS: all
- Browser: all

### PR DESCRIPTION
**Pull request by phanlezz**
# Remove .gz from Filenames

Issue: #2568 

## Description

Changed appearance of file names in the FinePanel: Files will now be display without file extension.
Specifically, all variations of "`.cc`, `.json`, `.gz` will be removed once from the end (must appear in that order) 
Updated Regex in `fileNameHelper.withoutCCExtension` to correctly remove the extensions mentioned above. 
Updated the use of `withoutCCExtension` in `util` files: `aggregationGenerator`, `customConfigHelper`, and `deltaGenerator`, so that extensions are no longer a part of compounded and generated names (like in the sidebar).

DeltaPreview

Closes #2568

### CODE REVIEW

**Comment by phanlezz** [comment]
> Please add a changelog entry. 🙂

I will do that, but I'm still not sure how to resolve the merge conflict. Should I resolve the issue on main and then merge? I'am not sure about the correct procedure 😟 .

**Review by Hall-Ma — APPROVED** [review]
Overall LGTM! I just found another case where we have still the file extension visible. Load a ccjson.gz file and open the attribute sidebar:

**Comment by phanlezz** [comment]
> Overall LGTM! I just found another case where we have still the file extension visible. Load a ccjson.gz file and open the attribute sidebar: [...] Maybe we could also remove the file extension?

Good catch, I will look into it!

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 
 ## [unreleased] (Added 🚀 | Changed | Removed 🗑 | Fixed 🐞 | Chore 👨‍💻 👩‍💻)
 
+### Changed
+
+-   Removed all mentions of regular `.cc` `.json` `.gz` file extensions from file panel and sidebar [#2793](https://github.com/MaibornWolff/codecharta/pull/2793)
```
Comment (shaman-apprentice) [inline CHANGELOG.md:12]: We use present tense in generell

```suggestion
-   Remove all mentions of regular `.cc` `.json` `.gz` file extensions from file panel and sidebar [#2793](https://github.com/MaibornWolff/codecharta/pull/2793)
```

File: visualization/app/codeCharta/util/fileNameHelper.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class FileNameHelper {
 		return fileName
 	}
 
-	static withoutCCJsonExtension(fileName: string) {
-		return fileName.replace(/(\.cc)?\.json?$/, "")
+	static withoutCCExtension(fileName: string) {
+		return fileName.replace(/(\.cc)?(\.json)?(\.gz)?$/, "")
```
Comment (shaman-apprentice) [inline visualization/app/codeCharta/util/fileNameHelper.ts:38]: Hadn't we agreed to only replace `(\.json)?(\.gz)?$`?

**Review by shaman-apprentice — APPROVED** [review]
[Optional] rework data in store to seperate file name and file extention, so that the pipe becomes obsolete. But as discussed current version is fine as well.

Eearly approve - please check my two comments.

File: visualization/app/codeCharta/util/fileNameHelper.ts (hunk 2 above)
Comment (phanlezz) [reply visualization/app/codeCharta/util/fileNameHelper.ts:38]: For now we will remove all extensions, like it is mentioned in the original issue #2568 . So old files are sill displayed correctly

**Pull request merged**